### PR TITLE
Improve components typing and remove `suppressImplicitAnyIndexErrors`

### PR DESCRIPTION
**Pull request by mickaelzhang**
## Screenshot / video

[if applicable for UI changes]

## What does this do?

[A short description of what the PR changes and why it's necessary]

- Give the reviewers context
- Is the impact global or localised?
- Any tech debt created or discovered?
- Include links to Jira ticket and Figma file if applicable

### CODE REVIEW

File: src/Button/Button.tsx [hunk 1]
```diff
@@ -45,7 +45,8 @@ type Props = {
   type?: 'button' | 'submit' | 'reset'
 }
 
-export type ButtonProps = Props & ButtonHTMLAttributes<HTMLButtonElement>
+export type ButtonProps = Props &
+  Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'color'>
```
Comment (mickaelzhang) [inline src/Button/Button.tsx:49]: `ButtonHTMLAttributes<HTMLButtonElement>` have a `color` type so it was overriding the one specified in `Props`

File: src/theme.ts [hunk 2]
```diff
@@ -36,6 +36,8 @@ const colors = {
   hometree: '#17171C',
 }
 
+export type Color = keyof typeof colors
```
Comment (Jmuskett) [inline src/theme.ts:39]: nice 😎

**Review by Jmuskett — APPROVED** [review]
LGTM 🚢

**Review by fleonard — APPROVED** [review]
finally Colours are typed! 🙌🏻

**Pull request merged**